apps/teams/models.py

```diff
@@ -61,11 +54,41 @@ class Membership(BaseModel):
         related_query_name="membership",
     )
 
+    class Meta:
+        abstract = True
+
+    def _get_permissions(self):
+        perm_cache_name = "_perm_cache"
+        if not hasattr(self, perm_cache_name):
+            perms = (
+                Permission.objects.filter(group__membership=self)
+                .values_list("content_type__app_label", "codename")
+                .order_by()
+            )
+            setattr(self, perm_cache_name, {"%s.%s" % (ct, name) for ct, name in perms})
+        return getattr(self, perm_cache_name)
+
+    def has_perm(self, perm):
+        return perm in self._get_permissions()
+
+    def has_perms(self, perm_list):
+        return all(self.has_perm(perm) for perm in perm_list)
+
+
+class Membership(BaseModel, PermissionsMixin):
+    """
+    A user's team membership
+    """
+
+    team = models.ForeignKey(Team, on_delete=models.CASCADE)
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    role = models.CharField(max_length=100, choices=roles.ROLE_CHOICES)
+
     def __str__(self):
         return f"{self.user}: {self.team}"
 
-    def is_admin(self) -> bool:
-        return self.role == roles.ROLE_ADMIN
+    def is_team_admin(self) -> bool:
+        return self.has_perms(["teams.change_team", "teams.delete_team"])
```

Comment:
Can't we ssimply check if the membership belongs to the `TEAM_ADMIN_GROUP` group?

Reply:
we could use that as a proxy but I thought it would be better to always check the permissions so that if we ever change the groups it still works.

Also this is how I found the issue with creating groups in the `post_migrate` signal.